tests/autocompletion/Earthfile

```diff
@@ -61,37 +59,44 @@ test-targets:
 +othertarget 
 +othertargetwithargs 
 +targetwithrequiredarg " > expected
-    RUN COMP_LINE="earthly +" COMP_POINT=9 earthly > actual
-    RUN diff expected actual
+    DO +COMPLETION_TEST --COMP_LINE="earthly +"
 
     RUN echo "+mytarget " > expected
-    RUN COMP_LINE="earthly +m" COMP_POINT=10 earthly > actual
-    RUN diff expected actual
+    DO +COMPLETION_TEST --COMP_LINE="earthly +m"
 
 test-target-with-build-args:
     COPY fake.earth ./Earthfile
 
     RUN echo "--city=
 --country=" > expected
-    RUN COMP_LINE="earthly +othertargetwithargs -" COMP_POINT=30 earthly > actual
-    RUN diff expected actual
+    DO +COMPLETION_TEST --COMP_LINE="earthly +othertargetwithargs -"
+    DO +COMPLETION_TEST --COMP_LINE="earthly +othertargetwithargs --c"
+
+    RUN echo "--city=" > expected
+    DO +COMPLETION_TEST --COMP_LINE="earthly +othertargetwithargs --ci"
+
+    RUN > expected
+    DO +COMPLETION_TEST --COMP_LINE="earthly +othertargetwithargs --city="
+    DO +COMPLETION_TEST --COMP_LINE="earthly +othertargetwithargs --city=foo"
+
+test-target-with-build-args-artifact-mode:
+    COPY fake.earth ./Earthfile
 
+    # autocomplete doesn't actually enforce artifact mode when offering completions
+    # for a target with an (assumed) artifact path
     RUN echo "--city=
 --country=" > expected
-    RUN COMP_LINE="earthly +othertargetwithargs --c" COMP_POINT=32 earthly > actual
-    RUN diff expected actual
+    DO +COMPLETION_TEST --COMP_LINE="earthly +othertargetwithargs/out -"
```

Comment:
Yeah... I did call that out in a comment ("Note that there's no enforcement that the --artifact flag was actually used."), but maybe I should have made that more visible.  While I knew that was 'wrong', I went with the rationale:
1. I wouldn't really consider it the auto-completions job to enforce the command is legit
2. This code path is probably relatively uncommon, since artifact path doesn't (can't) auto-complete.  So one would probably auto-complete the entire line and then backtrack to add the artifact path, or revisiti a prior earthly execution (i.e. bash history) that was already an artifact mode invocation, then modifying the build args (which is how I came across this issue)

Those two reasons combined is why I didn't think the added complexity of the additional code to track `--artifact`.  They aren't strong arguments, but figured it warranted KISS to start.

In regards to the first point... even now, if you do `earthly --artifact --im<TAB>`, it will autocomplete with `earthly --artifact --image`, which results in `Error: both image and artifact modes cannot be active at the same time`.  There's probably more occurrences like this.  It doesn't make your concern any less valid, but it would mean there are an number of outstanding 'bugs' like this.

If that doesn't sway you, I have no problem in making my change work 'correctly', even if the other invalid combinations don't get addressed (for a while at least)... let me know.

Reply:
> Then perhaps down the road, we could even do auto completion for the names of the artifacts (would get real difficult if there was any wildcards).

Good call... that itself is a legitimate reason to add the tracking (beyond the other 'softer' reasons).  It hadn't really occurred to me that (a subset of) artifacts _could_ be completed.  I'll incorporate your suggestions in the various comments later today.

Reply:
I believe my last commit addresses your concerns.  One thing that I didn't know is whether it's considered poor taste (or potential bug) to reorder/insert `const` entries like I did.  I know some languages can yield unpleasant surprises when using similar features.  But these were private, I didn't see them used in any mathematical operations numerical comparison, and the tests passed, so thought it was okay.

Reply:
reordering the consts (thus changing their binary values), is fine in this case -- we're not serializing them and passing them between different versions of earthly.